src/discover/matcher/google.js

```diff
@@ -166,44 +168,54 @@ export default class GoogleMatcher {
     }
   }
 
-  /**
-   * Test whether this class can handle an URL
-   *
-   * @param {URL} url url to match
-   * @param {Inventory} inventory
-   * @returns true if this class can handle the URL
-   */
-  static match(url, inventory) {
-    return inventory.getHostType(url.hostname) === 'google' || url.hostname.match(/^.*\.google\.com$/);
-  }
-
   /**
    * Extract some data from a URL to store in the inventory.
    *
-   * @param {import('../../index.js').AdminContext} context context
+   * @param {import('../support/AdminContext').AdminContext} context context
    * @param {URL} url url to extract data from
+   * @param {import('../inventory.js').InventoryEntry} entry entry
    * @returns object that contains additional entries to store in inventory
    */
-  static async extract(context, url, entry) {
+  async extract(context, url, entry) {
     const match = url.pathname.match(/\/.*\/folders\/([^?/]+)$/);
-    if (match) {
+    if (!match) {
+      return;
+    }
+
+    // eslint-disable-next-line no-param-reassign
+    [, entry.gdriveId] = match;
+    if (!entry.contentBusId) {
+      return;
+    }
+
+    // do not search for custom users in org/sites that
+    // are not listed in env.HLX_CUSTOM_GOOGLE_USERS
+    if (!this.customUserProjects.some((project) => project.match(entry))) {
+      return;
+    }
```

Comment:
noticed in reindex all, that we had an enormous amount of checks for custom users in every google backed project, although we won't include them when searching for google documents